### PR TITLE
Fixes #8306: Make rudder-dev modular to share code

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/8306

### CODE REVIEW

File: scripts/rudder-setup/Makefile [hunk 1]
```diff
@@ -9,5 +9,5 @@ ncf-setup: ncf-setup-src.sh *.sh
 	chmod +x $@
 
 clean:
-	rm -f rudder-setup
+	rm -f rudder-setup ncf-setup
```
Comment (jooooooon) [inline scripts/rudder-setup/Makefile]: This seems unrelated

File: scripts/rudder-dev/rudder-dev-src [hunk 2]
```diff
@@ -723,7 +526,7 @@ def update_nrm_users(project_id):
   if REDMINE_API_URL not in users:
     users[REDMINE_API_URL] = {}
   # just update it otherwise we would have only project specific data in it
-  user_list = requests.get(REDMINE_API_URL + "/projects/" + str(project_id) + "/memberships.json?limit=100", headers = {'X-Redmine-API-Key': REDMINE_TOKEN } )
+  user_list = requests.get(REDMINE_API_URL + "/projects/" + str(project_id) + "/memberships.json", headers = {'X-Redmine-API-Key': REDMINE_TOKEN } )
```
Comment (jooooooon) [inline scripts/rudder-dev/rudder-dev-src]: Looks like this change doesn't include some recent bugfixes!

File: scripts/rudder-dev/rudder-dev-src [hunk 3]
```diff
@@ -765,7 +568,7 @@ def ask_username(project_id, trigraph):
     sys.stdout.flush() # to display previous unfinished line
     trigraph = sys.stdin.readline().strip().upper()
     if trigraph == 'R':
-      user_ids = update_nrm_users(project_id)[REDMINE_API_URL]
+      update_nrm_users(project_id)
```
Comment (jooooooon) [inline scripts/rudder-dev/rudder-dev-src]: Also missing here

File: scripts/rudder-dev/rudder-dev-src [hunk 4]
```diff
@@ -249,209 +250,11 @@ ALT_CUSTOM_FIELD_PR = 1
...
-        if 'details' in journal:
-          for detail in journal['details']:
-            if detail['name'] == 'assigned_to_id' and 'old_value' in detail:
-              if int(detail['old_value']) != my_id:
-                info['last_assignee'] = int(detail['old_value'])
-  
-    self.info = info
-    return info
-
-  def branch_name(self):
```
Comment (jooooooon) [inline scripts/rudder-dev/rudder-dev-src]: This functions seems to have disappeared completely. It's no longer in the main file and I can't find it in any of the new files.

File: scripts/rudder-dev/rudder-dev-src [hunk 5]
```diff
@@ -1047,19 +728,7 @@ def create_pr(master_branch, info, message):
...
-    error_message = result['errors'][0]['message']
-    logfail("Error occured in create PR: " + error_message)
-  else:
-    logfail("Unkown error occured in create PR")
-
-  if not force:
-    exit(16)
-  else:
-    return None
+  return result['html_url']
```
Comment (jooooooon) [inline scripts/rudder-dev/rudder-dev-src]: This last block is also a recent bugfix that you're mising

File: scripts/rudder-dev/rudder-dev-src [hunk 6]
```diff
@@ -1660,7 +1328,7 @@ def merge_branch(old, new, strategy=None, automatic=False, name="", test=False):
     shell("git checkout -b " + test_branch)
   shell("git merge " + old + opts, "Merging " + name + " into " + new)
   first_merge = False
-  if not test:
+  if not Test:
```
Comment (jooooooon) [inline scripts/rudder-dev/rudder-dev-src]: This seems unrelated

File: scripts/rudder-setup/Makefile (hunk 1 above)
Comment (peckpeck) [inline scripts/rudder-setup/Makefile]: Yes, a minor line, i just stumbled upon it when copying the makefile.

**Comment by peckpeck** [comment]
PR rebased

**Comment by peckpeck** [comment]
Yes I'll have to redo the PR based on current master

**Comment by jooooooon** [comment]
I see you rebased but the bug fixes are still missing

**Comment by peckpeck** [comment]
Commit modified

File: scripts/rudder-dev/rudder-dev-src (hunk 4 above)
Comment (jooooooon) [inline scripts/rudder-dev/rudder-dev-src]: Did you find this function?

Comment (jooooooon) [inline scripts/rudder-dev/rudder-dev-src]: Ah, I see it is in Redmine.py now :)

**Comment by jooooooon** [comment]
Ah, I was ready to merge this, but it needs rebasing

**Comment by peckpeck** [comment]
PR rebased
